src/Connection.php

```diff
@@ -295,4 +295,18 @@ public function rollBack()
 
         return false;
     }
+
+    public $override_last_insert_id = null;
```

Comment:
How about ability to use callable function here and/or nested DSQL expression (sub-select) like `select xxx_seq.nextval from dual` ?

Reply:
I put this patch quickly together, probably there is a better implementation.

After this I started to wonder, why couldn't ID be passed in as a trigger?

And if during "insert" we somehow specify "ID" value, perhaps we shouldn't even attempt to get last_insert_id and simply set it to value of data[id_field]